src/providers/ethereum.ts

```diff
@@ -206,6 +272,48 @@ export default class EthereumProvider extends BloctoProvider implements Ethereum
           break;
         case 'eth_signTransaction':
         case 'eth_sendRawTransaction':
+          result = null;
+          break;
+        case 'wallet_addEthereumChain':
+          if (!payload?.params?.[0]?.chainId || !payload?.params?.[0]?.rpcUrls.length) {
+            throw new Error('Invalid params');
+          }
+          await getEvmSupport().then((supportNetwork) => {
+            this.supportNetwork = supportNetwork;
+            this.checkAndAddNetwork({
+              chainId: parseChainId(payload?.params?.[0]?.chainId),
+              rpcUrls: payload?.params?.[0].rpcUrls,
+            });
+          });
+          result = null;
+          break;
+        case 'wallet_switchEthereumChain':
+          if (!payload?.params?.[0]?.chainId) {
+            throw new Error('Invalid params');
+          }
+
+          if (!this.switchableNetwork[parseChainId(payload.params[0].chainId)]) {
+            const error: any = new Error(
+              'This chain has not been added to SDK. Please try wallet_addEthereumChain first.'
+            );
+            error.code = 4902;
```

Comment:
how does this 4902 be decided?

Reply:
Said by metamask but I also don't know why.
https://docs.metamask.io/guide/rpc-api.html#wallet-switchethereumchain

Reply:
Comment added